### PR TITLE
[Nest] Add missing null annotations and fix warnings

### PR DESCRIPTION
**Pull request by wborn**
Add missing null annotation to all binding packages except for `org.openhab.binding.nest.internal.data`.

I don't know if it is a good idea to also add null annotations to the Gson data classes. Currently the binding is very defensive and assumes everything could be null in these data classes. There are also unit tests to test it can handle all null values. Data could be null due to lack of API permissions or deprecated properties that may get removed some day.

So you might want to annotate all methods/fields with `@Nullable` but that seems quite laborious. There doesn't seem to be a `@NullByDefault` annotation for such classes. Also annotating the whole class with `@Nullable` doesn't work. 

@maggu2810 would it make sense to add null annotations to such data classes and if so how?

### CODE REVIEW

**Comment by maggu2810** [comment]
Classes that are constructed and filled by e.g. Gson are difficult to handle using nullness annotations as Gson uses reflection etc.

It would be possible to use a "isValid()" method on data classes that checks if the nullness requirements are fulfilled. The caller of the deserialization could call "isValid()" afterwards. If "isValid()" returns false, it can signal that error to its caller (e.g. notify the caller of a REST interface that some required values are missing). After that you can ensure that all the get methods or the public members fits to their nullness annotations and could be consumed in a safe manner.

For e..g Jackson it could be managed in a much better way, as you don't need to provide a default constructor but can use an constructor that arguments are annotated which JSON field should be used for that argument. This way you can control the construction in a much better way.

Another approach for such data classes / DTO is to keep all its members `@Nullable` and `public`.
The consumer itself needs to check that members are non null in front of its usage.
This could be done in e.g. a mapper that is called as soon as possible to create an "internal" class of that DTO.

With that options I assume the DTO to internal objects fits best what is done in the ESH code base.
Gson fills a DTO with nullable public members and a mapper is used to create an internal object. The mapper could check for non-null and all the code does not use the DTO itself but the mapped internal reference.

**Comment by wborn** [comment]
Thanks for your insights on this @maggu2810! It's difficult to determine what are valid objects when it is  allowed to give any subset of Nest API permissions to the Nest Binding. Currently the binding is very flexible with this and allows for only updating channels with state for which there is data.

I've also tried adding `@Nullable` to fields/getters/setters of the data objects. That results in many potential NPE warnings in the `hashCode()` and `equals(..)` methods of these objects.

So for now I think it is best to check that the binding can handle null values in data objects using the existing unit tests.

File: addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestStructureConfiguration.java [hunk 1]
```diff
@@ -8,14 +8,17 @@
...
  *
  * @author Wouter Born - Initial contribution
  * @author Wouter Born - Add device configuration to allow file based configuration
  */
+@NonNullByDefault
 public class NestStructureConfiguration {
     public static final String STRUCTURE_ID = "structureId";
     /** Structure ID which can be retrieved with the Nest API. */
-    public String structureId;
+    public String structureId = "";
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestStructureConfiguration.java]: Will / can it never be set to `null`?

File: addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestDeviceConfiguration.java [hunk 2]
```diff
@@ -8,14 +8,17 @@
...
  *
  * @author Wouter Born - Initial contribution
  * @author Wouter Born - Add device configuration to allow file based configuration
  */
+@NonNullByDefault
 public class NestDeviceConfiguration {
     public static final String DEVICE_ID = "deviceId";
     /** Device ID which can be retrieved with the Nest API. */
-    public String deviceId;
+    public String deviceId = "";
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestDeviceConfiguration.java]: Will / can it never be set to `null`?

File: addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestStructureConfiguration.java (hunk 1 above)
Comment (wborn) [reply addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestStructureConfiguration.java]: It's a required configuration parameter see the Nest [config.xml](https://github.com/wborn/openhab2-addons/blob/d3e2a278b8b101e4b98133b20ed088dac840321a/addons/binding/org.openhab.binding.nest/ESH-INF/config/config.xml#L42).

Without the `structureId` the handler would not know what data from updates should be used for state updates. For the same reason the Nest API needs this ID for handling commands.

I was doubting between annotating it with `@NonNullByDefault({})` and initializing it with an empty string. The empty string option looks cleaner and is guaranteed to prevent NPEs.

File: addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestDeviceConfiguration.java (hunk 2 above)
Comment (wborn) [reply addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestDeviceConfiguration.java]: It's a required configuration parameter see the Nest [config.xml](https://github.com/wborn/openhab2-addons/blob/d3e2a278b8b101e4b98133b20ed088dac840321a/addons/binding/org.openhab.binding.nest/ESH-INF/config/config.xml#L48).

File: addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestStructureConfiguration.java (hunk 1 above)
Comment (martinvw) [reply addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestStructureConfiguration.java]: Okay, great
